api/base/filters.py

```diff
@@ -577,7 +577,19 @@ def postprocess_query_param(self, key, field_name, operation):
             operation['source_field_name'] = 'provider___id'
 
         if field_name == 'id':
-            operation['source_field_name'] = 'guids___id'
+            values = operation['value']
+            object_ids = []
+
+            for val in values:
+                referent, version = Guid.load_referent(val)
+                if referent is None:
+                    continue
+                object_ids.append(referent.id)
+
+            # Override the operation to filter id__in=these object_ids
+            operation['source_field_name'] = 'id__in'
+            operation['value'] = list(object_ids)
+            operation['op'] = 'eq'
```

Comment:
Ah, make sense 👍 . We should move this to a helper method 

```python
self.postprocess_version_id_query_param(operation)
```

, similar to what we do for subjects

```python
if field_name == 'subjects':
    self.postprocess_subject_query_param(operation)
```